### PR TITLE
[No-5][Training]test-login-logout

### PR DESCRIPTION
**Pull request by loitq**
- Require laravel/dusk.
- Create .env.testing

### CODE REVIEW

**Comment by phantuan1803** [comment]
LGTM

**Review by minhtd92 — APPROVED** [review]
LGTM

File: .env.testing [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+AWS_DEFAULT_REGION=us-east-1
+AWS_BUCKET=
+
+PUSHER_APP_ID=
+PUSHER_APP_KEY=
+PUSHER_APP_SECRET=
+PUSHER_APP_CLUSTER=mt1
+
+MIX_PUSHER_APP_KEY="${PUSHER_APP_KEY}"
+MIX_PUSHER_APP_CLUSTER="${PUSHER_APP_CLUSTER}"
```
Comment (hienl) [inline .env.testing]: Please check which variable are use which are not , please remove unneccessary variable

File: tests/Browser/ExampleTest.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+     * @return void
+     */
+    public function testBasicExample()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/')
+                    ->assertSee('Laravel');
+        });
+    }
+}
```
Comment (hienl) [inline tests/Browser/ExampleTest.php]: Why have example in product code ?

File: tests/Browser/Pages/HomePage.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Assert that the browser is on the page.
+     *
+     * @param  \Laravel\Dusk\Browser  $browser
+     * @return void
+     */
+    public function assert(Browser $browser)
+    {
+        //
```
Comment (hienl) [inline tests/Browser/Pages/HomePage.php]: What is the test for this file ?

File: tests/Browser/Pages/HomePage.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    /**
+     * Get the element shortcuts for the page.
+     *
+     * @return array
+     */
+    public function elements()
```
Comment (hienl) [inline tests/Browser/Pages/HomePage.php]: why need new function that do exactly the same work with current siteElements() function ?

**Review by hienl — CHANGES_REQUESTED** [review]
Please Fix , what are you really test in this pull request ?

File: .env.testing (hunk 1 above)
Comment (loitq) [reply .env.testing]: i has removed code.

File: tests/Browser/ExampleTest.php (hunk 2 above)
Comment (loitq) [reply tests/Browser/ExampleTest.php]: this is code demo, i will be remove when code.

File: tests/Browser/Pages/HomePage.php (hunk 3 above)
Comment (loitq) [reply tests/Browser/Pages/HomePage.php]: it feature test for page... Temporarily unused

File: tests/Browser/Pages/HomePage.php (hunk 4 above)
Comment (loitq) [reply tests/Browser/Pages/HomePage.php]: it check element in page... but i don't use... i will remove file

**Review by hienl — APPROVED** [review]
LGTM

**Pull request merged**